java/google/registry/rde/imports/RdeImportUtils.java

```diff
@@ -184,4 +179,59 @@ public static Trid generateTridForImport() {
     return Trid.create(
         "Import_" + BaseEncoding.base64().encode(UUID.randomUUID().toString().getBytes()));
   }
+
+  public static BillingEvent.Recurring createAutoRenewBillingEventForDomainImport(
+      XjcRdeDomain domain, HistoryEntry historyEntry) {
+    final BillingEvent.Recurring billingEvent =
+        new BillingEvent.Recurring.Builder()
+            .setReason(Reason.RENEW)
+            .setFlags(ImmutableSet.of(Flag.AUTO_RENEW))
+            .setTargetId(domain.getRoid())
+            .setClientId(domain.getClID())
+            .setEventTime(domain.getExDate())
+            .setRecurrenceEndTime(END_OF_TIME)
+            .setParent(historyEntry)
+            .build();
+    return billingEvent;
+  }
+
+  public static PollMessage.Autorenew createAutoRenewPollMessageForDomainImport(
+      XjcRdeDomain domain, HistoryEntry historyEntry) {
+    final PollMessage.Autorenew pollMessage =
+        new PollMessage.Autorenew.Builder()
+            .setTargetId(domain.getRoid())
+            .setClientId(domain.getClID())
+            .setEventTime(domain.getExDate())
+            .setMsg("Domain was auto-renewed.")
+            .setParent(historyEntry)
+            .build();
+    return pollMessage;
+  }
+
+  public static HistoryEntry createHistoryEntryForDomainImport(XjcRdeDomain domain) {
+    XjcRdeDomainElement element = new XjcRdeDomainElement(domain);
+    final HistoryEntry historyEntry =
+        new HistoryEntry.Builder()
+            .setType(HistoryEntry.Type.RDE_IMPORT)
+            .setClientId(domain.getClID())
+            .setTrid(generateTridForImport())
+            .setModificationTime(DateTime.now())
```

Comment:
HistoryEntry modification times should always match the transaction time of the save - it makes it easier if need be to match the HistoryEntry log with other stuff that reflects the transaction time, including CreateAutoTimestamp/UpdateAutoTimestamp and the commit log system.  (In fact, HistoryEntry's modificationTime should arguably be a CreateAutoTimestamp; it predates the latter's existence.)

So here we should really pass in DateTime now as a parameter and provide ofy().getTransactionTime() from the callsite.

FWIW, also, in general our approach to time means it's typically a red flag to be using DateTime.now() except if you're explicitly not in a transaction context at all (like DomainInfoFlow).